src/main/java/com/exactpro/th2/crawler/Crawler.java

```diff
@@ -165,57 +183,84 @@ public Duration process() throws IOException, UnexpectedDataProcessorException {
             intervalStartForProcessor(dataProcessor, interval, state);
 
             Continuation continuation = state == null ? null : typeStrategy.continuationFromState(state);
-            DataParameters parameters = new DataParameters(info, crawlerId, sessionAliases);
-            CrawlerData<Continuation> data;
-            Report<Continuation> sendingReport = null;
-            long processedEvents = 0L;
-            Timestamp startTime = toTimestamp(interval.getStartTime());
-            Timestamp endTime = toTimestamp(interval.getEndTime());
+
+            Collection<String> newAliases = null;
```

Comment:
I would suggest to use `Collection.emptySet()` instead of `null`. It is always better to have the variable with not null value. Also, it might help to simplify the logic. For example, you can call `sessionAliases.addAll(newAliases);` right at the start and nothing will changes with `sessionAliases`